src/Notification/NotificationChannelRegistry.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Notification;
+
+class  NotificationChannelRegistry
+{
+    private array $channels;
+
+    public function __construct(iterable $channels)
+    {
+        $this->channels = $channels instanceof \Traversable ? iterator_to_array($channels) : $channels;
```

Comment:
This code instantiates all the channels, while only one of them is used.
Instead of tagged_iterator, it makes sense to use tagged_locator.